src/server/scripts/Northrend/zone_dalaran.cpp

```diff
@@ -199,30 +206,62 @@ class npc_minigob_manabonk : public CreatureScript
                     switch (eventId)
                     {
                         case EVENT_SELECT_TARGET:
+                        {
+                            std::vector<Player*> PlayerInDalaranList;
+                            GetPlayersInDalaran(PlayerInDalaranList);
+
+                            // Increases chance of event based on player count in Dalaran (100 players or more = 100% else player count%)
+                            if (PlayerInDalaranList.empty() || urand(1, 100) > PlayerInDalaranList.size())
+                                me->AddObjectToRemoveList();
+
                             me->SetVisible(true);
-                            DoCast(me, SPELL_TELEPORT_VISUAL);
-                            if (Player* player = SelectTargetInDalaran())
+                            DoCastSelf(SPELL_TELEPORT_VISUAL);
+                            if (Player* player = SelectTargetInDalaran(PlayerInDalaranList))
+                            {
+                                playerGuid = player->GetGUID();
+                                Position pos = player->GetPosition();
+                                float dist = frand(10.0f, 30.0f);
+                                float angle = frand(0.0f, 1.0f) * M_PI * 2.0f;
+                                player->MovePositionToFirstCollision(pos, dist, angle);
+                                me->NearTeleportTo(pos.GetPositionX(), pos.GetPositionY(), pos.GetPositionZ(), pos.GetOrientation());
+                            }
+                            events.ScheduleEvent(EVENT_LAUGH_1, Seconds(2));
+                            break;
+                        }
+                        case EVENT_LAUGH_1:
+                            me->HandleEmoteCommand(EMOTE_ONESHOT_LAUGH_NO_SHEATHE);
+                            events.ScheduleEvent(EVENT_WANDER, Seconds(3));
+                            break;
+                        case EVENT_WANDER:
+                            me->GetMotionMaster()->MoveRandom(8);
+                            events.ScheduleEvent(EVENT_PAUSE, Minutes(1));
+                            break;
+                        case EVENT_PAUSE:
+                            me->GetMotionMaster()->MoveIdle();
+                            events.ScheduleEvent(EVENT_CAST, Seconds(2));
+                            break;
+                        case EVENT_CAST:
+                            if (Player* player = me->GetMap()->GetPlayer(playerGuid))
                             {
-                                me->NearTeleportTo(player->GetPositionX(), player->GetPositionY(), player->GetPositionZ(), 0.0f);
                                 DoCast(player, SPELL_MANABONKED);
                                 SendMailToPlayer(player);
                             }
-                            events.ScheduleEvent(EVENT_BLINK, 3*IN_MILLISECONDS);
+                            else
+                            {
```

Comment:
just codestyle, remove this { } and merge xD

Reply:
Yeah I wasn't sure about that one. Generally if the "if" side is one line I go no braces both sides. But it looks ugly if you do:

if (blah)
{
    stuff();
    morestuff();
}
else
    differentstuff();

But, my opinion is all. I didn't check elsewhere how it's done.

Reply:
According to https://trinitycore.atlassian.net/wiki/spaces/tc/pages/2130103/C+Development+Standards#C++DevelopmentStandards-Brackets , "[ ... ] One line blocks does not need brackets". The example shows 2 options, one with brackets and one without for a single line after `else`. To me, that suggests you can choose to use brackets, although it is not needed.